packages/app-config-writer/src/smartlinks-config/generateSmartLinks.ts

```diff
@@ -0,0 +1,49 @@
+import { create as createStorage } from 'mem-fs';
+import { create } from 'mem-fs-editor';
+import type { Editor } from 'mem-fs-editor';
+import type { ToolsLogger } from '@sap-ux/logger';
+import type { SmartLinksConfig } from '..';
+import { getCredentials, getServices, writeSmartLinksConfig } from '.';
+
+/**
+ * @description Add smartlinks configuration to a UI5 application.
+ * @param basePath - the base path where the ui5-deploy/ui5.yaml is
+ * @param config - configuration of the smartlinks
+ * @param logger - logger
+ * @param fs - the memfs editor instance
+ * @returns Promise<Editor> - memfs editor instance with updated files
+ */
+export async function generateSmartLinksConfig(
+    basePath: string,
+    config: SmartLinksConfig,
+    logger?: ToolsLogger,
+    fs?: Editor
+): Promise<Editor> {
+    if (!fs) {
+        fs = create(createStorage());
+    }
+    if (!config.service) {
+        const service = (await getServices(basePath, logger, fs))[0];
+        const credentials = await getCredentials(service, logger);
+        config.service = { ...service, credentials };
+    }
```

Comment:
If you are at this stage, then you should already have all the data. In which case `config.service` is not present?

Reply:
This case is for the usage without prompts, if not chosen yet, the service still has to be determined.

Reply:
@johannes-kolbe There is no usage without prompts. In order to create the correct configuration we always need to prompt the user for the service and the credentials (if the credentials are not provided).